### ISSUE TITLE
Save and New in MetaModels lead to exception

### ISSUE THREAD

**Issue by seaneble** [contao-community-alliance/dc-general#86]
> Fatal error: Uncaught exception ContaoCommunityAlliance\DcGeneral\Exception\DcGeneralRuntimeException with message Unparsable encoded id value: '' thrown in composer/vendor/contao-community-alliance/dc-general/src/ContaoCommunityAlliance/DcGeneral/Contao/View/Contao2BackendView/IdSerializer.php on line 141

I have no idea whether this should rather be posted in MetaModels/core or here. Please close if this is the wrong place.

**Comment by Aybee** [comment]
Same here on **Contao 3.4.5** with `contao-community-alliance/dc-general dev-develop @ a69951`

On "Save and new" of an attribute. Bei "Speichern und neu" eines Attributes.

> Fatal error: Uncaught exception ContaoCommunityAlliance\DcGeneral\Exception\DcGeneralRuntimeException with message Unparsable encoded id value: '' thrown in system/modules/dc-general/classes/src/ContaoCommunityAlliance/DcGeneral/Contao/View/Contao2BackendView/IdSerializer.php on line 139

```
#0 system/modules/dc-general/classes/src/ContaoCommunityAlliance/DcGeneral/Contao/View/Contao2BackendView/Filter/LanguageFilter.php(70): ContaoCommunityAlliance\DcGeneral\Contao\View\Contao2BackendView\IdSerializer::fromSerialized('')
#1 system/modules/dc-general/classes/src/ContaoCommunityAlliance/DcGeneral/Contao/View/Contao2BackendView/Filter/LanguageFilter.php(50): ContaoCommunityAlliance\DcGeneral\Contao\View\Contao2BackendView\Filter\LanguageFilter->checkLanguage(Object(ContaoCommunityAlliance\DcGeneral\DefaultEnvironment))
#2 [internal function]: ContaoCommunityAlliance\DcGeneral\Contao\View\Contao2BackendView\Filter\LanguageFilter->handleAction(Object(ContaoCommunityAlliance\DcGeneral\Event\ActionEvent), 'dc-general.acti...', Object(Symfony\Component\EventDispatcher\EventDispatcher))
#3 system/modules/symfony-event-dispatcher/classes/EventDispatcher.php(168): call_user_func(Array, Object(ContaoCommunityAlliance\DcGeneral\Event\ActionEvent), 'dc-general.acti...', Object(Symfony\Component\EventDispatcher\EventDispatcher))
#4 system/modules/symfony-event-dispatcher/classes/EventDispatcher.php(53): Symfony\Component\EventDispatcher\EventDispatcher->doDispatch(Array, 'dc-general.acti...', Object(ContaoCommunityAlliance\DcGeneral\Event\ActionEvent))
#5 system/modules/dc-general/classes/src/ContaoCommunityAlliance/DcGeneral/Controller/DefaultController.php(121): Symfony\Component\EventDispatcher\EventDispatcher->dispatch('dc-general.acti...', Object(ContaoCommunityAlliance\DcGeneral\Event\ActionEvent))
#6 system/modules/metamodels/classes/src/MetaModels/BackendIntegration/Module.php(77): ContaoCommunityAlliance\DcGeneral\Controller\DefaultController->handle(Object(ContaoCommunityAlliance\DcGeneral\Action))
#7 system/modules/core/classes/Backend.php(406): MetaModels\BackendIntegration\Module->generate()
#8 system/modules/core/controllers/BackendMain.php(126): Contao\Backend->getBackendModule('metamodels')
#9 contao/main.php(20): Contao\BackendMain->run()
#10 {main}
```

**Comment by Aybee** [comment]
MetaModels/core or contao-community-alliance/dc-general? https://github.com/MetaModels/core/issues/500

**Comment by discordier** [comment]
No feedback, closing therefore.

**Comment by seaneble** [comment]
GitHub does not notify when you're only applying the "testing" label. You need to add a comment as well, or so it seems. Nobody had any idea you need feedback :smile:

**Comment by tim-bec** [comment]
Hm yeah, thats the part minimodel should take response of. but where is she @discordier ?

**Comment by seaneble** [comment]
I did some tests with the newest version available via Composer and did not receive any unusal messages with "Save and new".

Unfortunately, nothing happens, either. The button behaves like "Save and close". No new item is created.

**Comment by discordier** [comment]
@tim-bec She is only responsible for the MetaModels organization so far and does not give a damn about dc-general therefore.

@seaneble As long as there is no bug, the missing feature will be added by @dmolineus when he returns from holiday I hope.

**Comment by seaneble** [comment]
Well, as there is no Exception, one could try to avoid the term "bug"…

**Comment by discordier** [comment]
Exactly, there was a bug (exception) but is no more. Now we have the missing feature that save&new does not redirect to an empty edit mask.

**Comment by dmolineus** [comment]
It actually should be implemented since 581c9c76c79c742143a3d4a101239e81961c5f30. See https://github.com/contao-community-alliance/dc-general/blob/develop/src/ContaoCommunityAlliance/DcGeneral/Contao/View/Contao2BackendView/EditMask.php#L517-L523

Maybe there's an issue with some sorting definitions. Could you please tell me which sorting option is defined and used? I will have a look on it.

**Comment by MacKP** [comment]
I would give you my options I use in my MM.. but unfortiunaly I get an error, when I klick on the Sorting ->
```
Fatal error: Call to a member function getName() on a non-object in /.../composer/vendor/contao-community-alliance/meta-palettes/src/MetaPalettesBuilder.php on line 394
```
(Contao 3.4.5 and MM from yesterday)

I think I have just order by Title 'Text' and order by PLZ 'Number'. Just the first Letter and Numbler.

regards

**Comment by seaneble** [comment]
I'm a bit confused. How do sorting and "Save and new" fit together? Are these the same or separate issues?

**Comment by dmolineus** [comment]
If you use a manual sorting you have to choose where to insert the new item at the moment. If no manual sorting in created you don't have to worry about it.

At the moment the new item is not automatically added after the current one (as Contao standard).

**Comment by discordier** [comment]
@MacKP Your issue should be fixed with meta-palettes 1.9.6. There was a bug in there.

**Comment by MacKP** [comment]
@discordier jep works for me now. My setting is like i sayed before. @dmolineus when you need more Information feel free to contact me ;-)

**Comment by dmolineus** [comment]
It should be working now after #148 got merged. The current behaviour is:
 * For non manual sorting: Redirect to the input mask.
 * For manual sorting: Jump to list/tree/parent view and show the paste button.

The implemented behaviour could be improved to automatically guess the manual sorting position. But the feature should work as described above.

@MacKP Would be great if you test it again and confirm everything is working. This fix is included in "dcg 2.0.0-beta26". 

Note: If you run a composer update and install the latest metamodels/core you have to install the latest dev version of the dc-general. The latest metamodels/core contains a breaking change (https://github.com/MetaModels/core/pull/824#issuecomment-127168764).

**Comment by MacKP** [comment]
I am working at a new Projekt right now. right now it works for Attributes (i can test items later). Works for me. But i have one question: Why going back to the list/tree/parent view for manual sorting (i asume thats why this is the same there in the Attributes). Everywhere in Contao it would paste the new item after the item i made before. Would be one klick less ;-)

**Comment by dmolineus** [comment]
It mainly has technical reason. I only fixed the issues which prevents the save+new from working. Gonna change it so that it gonna work as known in Contao.

When setting up a site structure in Contao I quite often want the behaviour as being currently implemented in the dcg.

**Comment by MacKP** [comment]
Items with:
- no grouping
- textattribute
- asc
- no manual sorting
Don't work with save and close. I get the clipboard, but no arrow to paste it.

**Comment by dmolineus** [comment]
Some changes in the codebase reintroduced the issue and gonna be fixed with #151.

**Comment by xantippe** [comment]
"Speichern und Neu" getestet
* MM
* Attribute
* Render-Einstellungen
* Eingabemasken
* Filter

kann m.E. zu
